src/Configuration.php

```diff
@@ -129,18 +154,45 @@ public function getDOMDocument() {
      * @param  string     $xpathExpression
      * @param  boolean    $alwaysReturnArray
      * @return array|null
-     **/
+     * */
     public function get($xpathExpression, $alwaysReturnArray = false) {
+        if ($this->useCaching && $alwaysReturnArray) {
+            if (!array_key_exists($xpathExpression, $this->cache["alwaysArray"])) {
+                $this->cache["alwaysArray"][$xpathExpression] = $this->getConfiguration($xpathExpression, $alwaysReturnArray);
+            }
+            return $this->cache["alwaysArray"][$xpathExpression];
+        } elseif ($this->useCaching) {
+            if (!array_key_exists($xpathExpression, $this->cache["optionalArray"])) {
+                $this->cache["optionalArray"][$xpathExpression] = $this->getConfiguration($xpathExpression, $alwaysReturnArray);
+            }
+            return $this->cache["optionalArray"][$xpathExpression];
+        } else {
+            return $this->getConfiguration($xpathExpression, $alwaysReturnArray);
+        }
+    }
+
+    /**
+     * getConfiguration
+     *
+     * Returns an array representation of the XML nodes from the requested $xpathExpression
+     *
+     * @api
+     *
+     * @access private
+     * @param  string     $xpathExpression
+     * @param  boolean    $alwaysReturnArray
+     * @return array|null
+     * */
+    private function getConfiguration($xpathExpression, $alwaysReturnArray = false) {
```

Comment:
getConfiguration sort of implies that the entire configuration is returned. Better to change this to something like getFromDOMDocument or something.